raiden/network/rpc/client.py

```diff
@@ -167,13 +167,14 @@ def geth_discover_next_available_nonce(
     # the user, these will be the younger transactions. Because this needs the
     # largest nonce, queued is checked first.
 
+    address = to_checksum_address(address)
```

Comment:
The absence of this is why we did not see this in real life and why you only saw this from the SP. And this was yet another bug.

Web3 returns an `AttributeDict` that looks like this:

```
KeysView(AttributeDict({'10':
'0x50607A6053295b4d7E3A4F38242c5C0a3e47793a: 0 wei + 50000 gas
1000000000 wei', '100': '0x2C3154203f5127552664763D3053343d2f756151: 0
wei + 50000 gas 1000000000 wei',
```

And its addresses are not only hex, but they are also checksummed. As
such we need to also use checksummed addressed in this query or else
we will never get a correct response from geth.